### PR TITLE
Initial support for formatted FileRead()

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
https://github.com/certik/fastGPT/tree/lf28run

### CODE REVIEW

**Review by certik — APPROVED** [review]
This seems related to: https://github.com/lfortran/lfortran/pull/2330

**Comment by certik** [comment]
I think this PR works fastGPT!

**Comment by Shaikh-Ubaid** [comment]
This PR gets chat program in the `lf28run` branch at fastgpt (https://github.com/certik/fastGPT/tree/lf28run) to run. It currently works for first 2 prompts but segfaults on the exact third run.

**Comment by certik** [comment]
Already main seems to work with integer(1) -> character.

**Comment by Shaikh-Ubaid** [comment]
> first 2 prompts but segfaults on the exact third run.

It segfaults exactly when there is call to the `ffn()` inside the `transformer_block()`. It might again be related to array allocation on stack.

File: integration_tests/file_10.f90 [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    implicit none
+
+    character(1024) :: tmp
+    character(:), allocatable :: sentence1
+    character(:), allocatable :: sentence2
+    character(:), allocatable :: sentence3
+    character(:), allocatable :: sentence4
+    integer :: ios, u = 11
+
+    open(u, file="file_09_data.txt", form="formatted", access="stream", status="old")
```
Comment (certik) [inline integration_tests/file_10.f90:11]: Should this be `file_10_data.txt`?

File: integration_tests/file_10.f90 [hunk 2]
```diff
@@ -0,0 +1,33 @@
+program file_09
```
Comment (certik) [inline integration_tests/file_10.f90:1]: `file_10`

File: integration_tests/file_10.f90 (hunk 1 above)
Comment (Shaikh-Ubaid) [reply integration_tests/file_10.f90:11]: Yes

**Review by certik — APPROVED** [review]

**Review by czgdp1807 — APPROVED** [review]

**Pull request merged**